frontend/web/components/modals/CreateFlag.js

```diff
@@ -1214,68 +1214,71 @@ const CreateFlag = class extends Component {
-                                                                !name ||
-                                                                invalid ||
-                                                                !savePermission ||
-                                                                (manageSegmentOverridesEnabled &&
-                                                                  !manageSegmentsOverrides)
-                                                              }
-                                                            >
-                                                              {isSaving
-                                                                ? 'Updating'
-                                                                : 'Update Segment Overrides'}
-                                                            </Button>
-                                                          )
-                                                        }}
-                                                      </Permission>
-                                                    </div>,
-                                                  )
-                                                }
-                                              </Permission>
+                                              <div className='text-right'>
+                                                <Permission
+                                                  level='environment'
+                                                  permission={Utils.getManageFeaturePermission(
+                                                    is4Eyes,
+                                                    identity,
+                                                  )}
+                                                  id={this.props.environmentId}
+                                                >
+                                                  {({
+                                                    permission: savePermission,
+                                                  }) => (
+                                                    <Permission
+                                                      level='environment'
+                                                      permission={
+                                                        'MANAGE_SEGMENT_OVERRIDES'
+                                                      }
+                                                      id={
+                                                        this.props.environmentId
+                                                      }
+                                                    >
+                                                      {({
+                                                        permission:
+                                                          manageSegmentsOverrides,
+                                                      }) => {
```

Comment:
I don't quite understand what this code is doing but I have tested it functionally and it seems to behave how I expect!